design/KEPS/kiali-enhancement-proposal/proposal.md

```diff
@@ -0,0 +1,57 @@
+
+## Goals
+
+- Provide a natural way for features to progress from github discussion(s) to an implementable proposal.
+- Make it easier to formally present a large enhancement for the project, receive feedback, and reach a consensus on future implementation.
+- Versioned and easily searchable proposals.
+- Public process.
+- Establish owners and a roadmap for the proposal.
+
+## Non-goals
+
+- Impose a strict process for all changes (large or small) to adhere to. The KEP process is meant to enhance the speed at which large features are implemented and improve refinement for those features by making it easier to collaborate on the design of those features before they are implemented. It is not meant to impose a burden on making changes to the project.
+- Replace github discussions. It's expected that most large features will first start as an informal discussion which will then lead to a formal proposal.
+
+# Solution
+
+A `design/KEPS` dir is created in the [kiali repo](https://github.com/kiali/kiali) where all KEPS will be submitted and stored.
+
+The KEP process will be as follows:
+
+1. Proposal owner submits a proposal for an enhancement by opening a PR to the Kiali github repo. The PR contains a single markdown file describing the proposal using this KEP as a template. The PR includes any additional materials such as diagrams, pictures, videos etc. Both the proposal and supporting materials are organized under a single directory in the `design/KEPS` section of the kiali repo. As an example, this proposal is submitted under the `design/KEPS/kiali-enhancement-proposal/` directory.
+2. Owner assigns `kiali/maintainers` as a reviewer for the PR and assigns themself and any other owners as an `Assignee`.
+3. Maintainers provide timely feedback to the proposal in the form of github reviews on the PR.
+4. Proposal owners respond to feedback and update proposal by adding commits to the github PR.
+5. (Optional) Any pending proposals are discussed and presented during the end of sprint meetings.
+6. After feedback has been sufficiently addressed, proposals are either accepted or rejected by Kiali maintainers.
+7. Proposal is broken down into epics/issues and implemented. The roadmap for the proposal is updated along the way.
+
+It is expected that both proposal owners and maintainers will be active participants in any open proposals by giving and responding to feedback. Proposals should strive to reach consensus by the end of the next sprint.
```

Comment:
> Proposals should strive to reach consensus by the end of the next sprint.

I have feeling that set a deadline to have a consensus may introduce pressure, I'd propose to reword this to have "progress".

What do I mean ? I'd like if a proposal is quite active even it hasn't reached any consensus.
I.e.:
- There is a first design 1 proposed by folk A
- Folk B, reviews and proposes a design 2, reviewed by Folk A.
- Iteration over the previous step, even if there is no consensus, but there is progress.

It may involve more than a Sprint.

In general, a proposal shouldn't take too long, but I have the impression that there are areas that may need time if we want to have a good proposal, discussion and consensus.

Examples:
- Redesign of APIs (introducing GraphQL instead of pure REST).
- Redesign of new backend model.
- Discussion about what is the scope of a service (health) and how these services depend on others internal components.
- Push vs pull model in the UI.

Just to cite some areas that are suitable for KEP but I they deserves a detailed discussion.

Reply:
Personally I guess that time boxing is something that add pressure and makes that if the last day of the deadline we can rush. 
I think that the commitment should be that maintainers should get the commitment to keep active KEP until that is resolved.
To me the "active" verb is more flexible (it should be stalled) and also doesn't box something to rush that in a sprint there is a decision.
Also, in case that there are any need to poll, the governance is still there.
I'd try to avoid to put more time boxing rather than the formal one defined in the governance doc.

I think that when there is a relevant KEP proposed is in the maintainers interestest to participate, and also in the owners to seek consensus.

Reply:
I'd say the goal is less to add pressure and more to set the expectation that the proposals should have active involvement and eventually resolve instead of staying open forever. I'm ok with removing the timeboxing but perhaps we can also put a cap on the number of active proposals there are at any given time like what @jshaughn [suggested](https://github.com/kiali/kiali/discussions/4208#discussioncomment-1084133)?

Reply:
+1 saying that the number of active proposals can be limited and others may be queue is a right topic, to answer a question "why my proposal is not yet started", "because, mantainers are reviewing A,B,C"

Reply:
Alright I didn't propose a hard limit on the number of active proposals but merely stated that priority would be given to existing proposals over new ones.